scripts/check_delta.py

```diff
@@ -0,0 +1,87 @@
+            description = "find different tests delta")
+    parser.add_argument("-x", "--xml-file", default = None,
+            action = "append", help = "xml file")
+
+    return parser.parse_args()
+
+def parse_xml(xmlfiles):
+    # get all test cases name
+    tree_1 = ET.parse(xmlfiles[0])
+    testsuite_1 = tree_1.getroot()[0]
+    testcase_1 = testsuite_1.findall("testcase")
+    testcase_name_1 = set()
+    for item in testcase_1:
+        testcase_name_1.add(item.attrib["name"])
+
+    tree_2 = ET.parse(xmlfiles[1])
+    testsuite_2 = tree_2.getroot()[0]
+    testcase_2 = testsuite_2.findall("testcase")
+    testcase_name_2 = set()
+    for item in testcase_2:
+        testcase_name_2.add(item.attrib["name"])
+
+    # get all common test case name
+    common_test_name = testcase_name_1.intersection(testcase_name_2)
+
+    # remove commont test case
+    for item in testcase_1:
+        if item.attrib["name"] in common_test_name:
+            testsuite_1.remove(item)
+            testsuite_1.attrib["tests"] = str(int(testsuite_1.attrib["tests"]) - 1)
+            if item.find("skipped") is not None and item.find("skipped").attrib["type"] == "skipped":
+                testsuite_1.attrib["skip"] = str(int(testsuite_1.attrib["skip"]) - 1)
+            elif item.find("errors") is not None and item.find("errors").attrib["type"] == "errors":
+                testsuite_1.attrib["errors"] = str(int(testsuite_1.attrib["errors"]) - 1)
+            elif item.find("failures") is not None and item.find("failures").attrib["type"] == "failures":
+                testsuite_1.attrib["failures"] = str(int(testsuite_1.attrib["failures"]) - 1)
+            else:
+                pass
+
+    for item in testcase_2:
+        if item.attrib["name"] in common_test_name:
+            testsuite_2.remove(item)
+            testsuite_2.attrib["tests"] = str(int(testsuite_2.attrib["tests"]) - 1)
+            if item.find("skipped") is not None and item.find("skipped").attrib["type"] == "skipped":
+                testsuite_2.attrib["skip"] = str(int(testsuite_2.attrib["skip"]) - 1)
+            elif item.find("errors") is not None and item.find("errors").attrib["type"] == "errors":
+                testsuite_2.attrib["errors"] = str(int(testsuite_2.attrib["errors"]) - 1)
+            elif item.find("failures") is not None and item.find("failures").attrib["type"] == "failures":
+                testsuite_2.attrib["failures"] = str(int(testsuite_2.attrib["failures"]) - 1)
+            else:
+                pass
+
+    # no need for time info
+    testsuite_1.attrib.pop("time")
+    testsuite_2.attrib.pop("time")
+
+    # write unique tests into a new xml file
+    if os.path.exists("./xml_report"):
+        shutil.rmtree("./xml_report")
+    os.mkdir("./xml_report")
+
+    # store into a new xml file
+    tree_1.write("./xml_report/" + os.path.basename(xmlfiles[0]))
+    tree_2.write("./xml_report/" + os.path.basename(xmlfiles[1]))
```

Comment:
not understanding what you are doing here. writing 2 files? the delta should be one file, no?

Reply:
because i want for each xml, only store his different tests of him selves, then i can use junit2html --report-matrix to generate a html format to show tests delta between two boards or two commits.